downstairs/src/region.rs

```diff
@@ -965,6 +1106,162 @@ impl Region {
         Ok(())
     }
 
+    /**
+     * Repair an extent from another downstairs
+     *
+     * We need to repair an extent in such a way that an interruption
+     * at any time can be recovered from.
+     *
+     * Let us assume we are repairing extent 012
+     *  1. Make new 012.copy dir  (extent name plus: .copy)
+     *  2. Get all extent files from source side, put in 012.copy directory
+     *  3. fsync files we just downloaded
+     *  4. Rename 012.copy dir to 012.replace dir
+     *  5. fsync extent directory ( 00/000/ where the extent files live)
+     *  6. Replace current extent 012 files with copied files of same name
+     *    from 012.replace dir
+     *  7. Remove any files in extent dir that don't exist in replacing dir
+     *  8. fsync files after copying them (new location).
+     *  9. fsync containing extent dir
+     * 10. Rename 012.replace dir to 012.completed dir.
+     * 11. fsync extent dir again (so dir rename is persisted)
+     * 12. Delete completed dir.
```

Comment:
It probably would not hurt to do it, as if things crash after the delete the directory could come back, but in that scenario we will just delete it again.  And, I actually have an XXX to fsync one last time, so when the
fsync changes come in (probably next repair PR) It will be done!